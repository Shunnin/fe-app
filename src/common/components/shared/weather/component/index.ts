export { Temperature } from './temperature/temperature.component';
export { WeatherCard } from './weather-card/weather-card.component';
export { WeatherItem } from './weather-item/weather-item.component';
export * from './weather-icon';
