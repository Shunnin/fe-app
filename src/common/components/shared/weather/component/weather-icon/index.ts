export { WeatherIcon } from './weather-icon.component';
export * from './weather-icon.constant';
