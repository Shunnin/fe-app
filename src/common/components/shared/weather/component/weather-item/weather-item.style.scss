@import 'style/foundation/config';

.weather-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: $b-space;

  & &__day {
    font-weight: $b-font-weight-bold;
    font-size: $b-font-size;
    color: $c-lb7;
  }

  svg {
    margin: $b-space-s 0;
  }

  & &__forecast {
    font-weight: $b-font-weight-medium;
    font-size: $b-font-size-sm;
    color: $c-gr1;
  }

  & &__degree {
    font-size: $b-font-size-xs;
    color: $c-lb7;
    width: 5rem;
    text-align: center;
  }
}
