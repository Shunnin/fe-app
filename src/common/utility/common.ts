export const getNextSevenDays = (): string[] => {
  const days = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
  const next7Days = [];

  // eslint-disable-next-line no-magic-numbers
  for (let i = 0; i < 7; i++) {
    // eslint-disable-next-line no-magic-numbers
    next7Days.push(days[new Date(Date.now() + (i + 1) * 24 * 60 * 60 * 1000).getDay()]);
  }

  return next7Days;
};
