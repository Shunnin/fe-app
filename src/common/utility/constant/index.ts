export * from './number';
export * from './common';
export * from './url';
