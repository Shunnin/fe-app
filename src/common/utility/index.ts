export * from './route';
export * from './constant';
export * from './promise';
export * from './temperature';
export * from './common';
export * from './url';
