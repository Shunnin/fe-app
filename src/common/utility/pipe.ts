export const pipe =
  (...hofs) =>
  x =>
    hofs.reduceRight((currying, hof) => hof(currying), x);
