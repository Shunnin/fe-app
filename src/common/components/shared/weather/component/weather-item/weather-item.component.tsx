import { FC } from 'react';

import { Temperature } from '../temperature/temperature.component';
import { WeatherIcon } from '../weather-icon';

import './weather-item.style';

interface IWeatherItemProps {
  day: string;
  weatherCode: number;
  tempMax: number;
  tempMin: number;
  forecast: string;
}

export const WeatherItem: FC<IWeatherItemProps> = ({ day, weatherCode, tempMax, tempMin, forecast }) => {
  return (
    <div className="weather-item">
      <div className="weather-item__day">{day}</div>
      <WeatherIcon code={weatherCode} />
      <div className="weather-item__forecast">{forecast}</div>
      <span className="weather-item__degree">
        <Temperature value={tempMax} />
        <small>&nbsp;/&nbsp;</small>
        <Temperature value={tempMin} />
      </span>
    </div>
  );
};
