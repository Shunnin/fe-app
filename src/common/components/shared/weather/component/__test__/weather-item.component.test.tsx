import { render, screen } from '@testing-library/react';

import { createMockStore, withReduxProvider } from '../../../../../utility/test';

import { WeatherItem } from '../weather-item/weather-item.component';

describe('<WeatherItem />', () => {
  let store = null;

  const props = {
    day: 'Tue',
    weatherCode: 803,
    tempMax: 33,
    tempMin: 25,
    forecast: 'Sunny',
  };

  beforeAll(() => {
    store = createMockStore();
  });

  it('renders weather item and matches the snapshot when given valid weather', () => {
    const { container } = render(withReduxProvider(store)(<WeatherItem {...props} />));

    const weatherDay = screen.getByText(props.day);
    const weatherForecast = screen.getByText(props.forecast);

    expect(container.firstChild).toMatchSnapshot();
    expect(weatherDay).toBeInTheDocument();
    expect(weatherForecast).toBeInTheDocument();
  });
});
