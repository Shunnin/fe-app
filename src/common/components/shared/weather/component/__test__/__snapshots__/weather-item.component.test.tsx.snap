// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherItem /> renders weather item and matches the snapshot when given valid weather 1`] = `
<div
  class="weather-item"
>
  <div
    class="weather-item__day"
  >
    Tue
  </div>
  <reactcomponent />
  <div
    class="weather-item__forecast"
  >
    Sunny
  </div>
  <span
    class="weather-item__degree"
  >
    <span
      class="temperature"
    >
      33
      <sup>
        °
      </sup>
    </span>
    <small>
       / 
    </small>
    <span
      class="temperature"
    >
      25
      <sup>
        °
      </sup>
    </span>
  </span>
</div>
`;
